Civi/Core/Themes.php

```diff
@@ -0,0 +1,283 @@
+ | Copyright CiviCRM LLC (c) 2004-2016                                |
+ +--------------------------------------------------------------------+
+ | This file is a part of CiviCRM.                                    |
+ |                                                                    |
+ | CiviCRM is free software; you can copy, modify, and distribute it  |
+ | under the terms of the GNU Affero General Public License           |
+ | Version 3, 19 November 2007 and the CiviCRM Licensing Exception.   |
+ |                                                                    |
+ | CiviCRM is distributed in the hope that it will be useful, but     |
+ | WITHOUT ANY WARRANTY; without even the implied warranty of         |
+ | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.               |
+ | See the GNU Affero General Public License for more details.        |
+ |                                                                    |
+ | You should have received a copy of the GNU Affero General Public   |
+ | License and the CiviCRM Licensing Exception along                  |
+ | with this program; if not, contact CiviCRM LLC                     |
+ | at info[AT]civicrm[DOT]org. If you have questions about the        |
+ | GNU Affero General Public License or the licensing of CiviCRM,     |
+ | see the CiviCRM license FAQ at http://civicrm.org/licensing        |
+ +--------------------------------------------------------------------+
+ */
+
+namespace Civi\Core;
+
+use Civi;
+
+/**
+ *
+ * @package CiviCRM_Hook
+ * @copyright CiviCRM LLC (c) 2004-2016
+ */
+class Themes {
+
+  /**
+   * The "default" theme adapts based on the latest recommendation from civicrm.org
+   * by switching to DEFAULT_THEME at runtime.
+   */
+  const DEFAULT_THEME = 'greenwich';
+
+  /**
+   * Fallback is a pseudotheme which can be included in "search_order".
+   * It locates files in the core/extension (non-theme) codebase.
+   */
+  const FALLBACK_THEME = '_fallback_';
+
+  const PASSTHRU = 'PASSTHRU';
+
+  /**
+   * @var string
+   *   Ex: 'judy', 'liza'.
+   */
+  private $activeThemeKey = NULL;
+
+  /**
+   * @var array
+   *   Array(string $themeKey => array $themeSpec).
+   */
+  private $themes = NULL;
+
+  /**
+   * @var \CRM_Utils_Cache_Interface
+   */
+  private $cache = NULL;
+
+  /**
+   * Theme constructor.
+   * @param \CRM_Utils_Cache_Interface $cache
+   */
+  public function __construct($cache = NULL) {
+    $this->cache = $cache ? $cache : Civi::cache('long');
+  }
+
+  /**
+   * Determine the name of active theme.
+   *
+   * @return string
+   *   Ex: "greenwich".
+   */
+  public function getActiveThemeKey() {
+    if ($this->activeThemeKey === NULL) {
+      // Ambivalent: is it better to use $config->userFrameworkFrontend or $template->get('urlIsPublic')?
+      $config = \CRM_Core_Config::singleton();
+      $settingKey = $config->userFrameworkFrontend ? 'theme_frontend' : 'theme_backend';
```

Comment:
@totten @eileenmcnaughton one of the issues i raised in the PR from Agileware was that userFrameworkFrontend isn't set in Drupal integration (and likely not in backdrop either) but in Wordpress and Joomla

Reply:
@seamuslee001 the way the form works is that the 'theme_frontend' setting is not exposed for drupal users (since it's not used) - I think that's fine at this stage. I think we could discuss further if drupal needs are met but I would prefer that as a follow up